OpenRA.Game/IUtilityCommand.cs

```diff
@@ -9,12 +9,43 @@
  */
 #endregion
 
+using System;
+using System.Reflection;
+using OpenRA.Primitives;
 using OpenRA.Traits;
 
 namespace OpenRA
 {
 	public class Utility
 	{
+		static readonly ConcurrentCache<Type, FieldInfo[]> TypeFields =
+			new ConcurrentCache<Type, FieldInfo[]>(type => type.GetFields());
+
+		static readonly ConcurrentCache<(MemberInfo Member, Type AttributeType), bool> MemberHasAttribute =
+			new ConcurrentCache<(MemberInfo Member, Type AttributeType), bool>(
+				x => Attribute.IsDefined(x.Member, x.AttributeType));
+
+		static readonly ConcurrentCache<(MemberInfo Member, Type AttributeType, bool Inherit), object[]> MemberCustomAttributes =
+			new ConcurrentCache<(MemberInfo Member, Type AttributeType, bool Inherit), object[]>(
+				x => x.Member.GetCustomAttributes(x.AttributeType, x.Inherit));
```

Comment:
Do these use up memory even when running the game? (not the utility)

Reply:
No, because I've used this cache only for utility commands, the regular game code never touches the `Utility` class and so won't be affected by it.